cmake/BLTMacros.cmake

```diff
@@ -1363,6 +1363,134 @@ macro(blt_print_target_properties)
 endmacro(blt_print_target_properties)
 
 
+##------------------------------------------------------------------------------
+## blt_print_variables([NAME_REGEX  <regular_expression_string>]
+##                     [VALUE_REGEX <regular_expression_string>]
+##                     [IGNORE_CASE])
+##
+## Prints all (or filtered) variables and their values in the current scope.
+## 
+## Defaults:
+## The regexes are case-sensitive unless the IGNORE_CASE option is supplied
+## NAME_REGEX = .* (print every property name)
+## VALUE_REGEX = .* (print every property value)
+## 
+##------------------------------------------------------------------------------
+macro(blt_print_variables)
+
+    set(options            IGNORE_CASE)
+    set(singleValuedArgs   NAME_REGEX VALUE_REGEX)
+    set(multiValuedArgs)
+
+    # parse and initialize arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+
+    message(STATUS "[blt_print_variables] The following variables are defined at the calling site in '${CMAKE_CURRENT_LIST_FILE}' -- ")
```

Comment:
"calling site" or maybe "directory scope" better here?

Reply:
Thanks. I wasn't crazy about "calling site", but wanted to get across the idea that the results depend on where you call it since it knows about the variables that are in scope when you call it.

I don't think "directory scope" captures it though.